File: problem_2/even_fibonacci.rb
# frozen_string_literal: true

first = 1
second = 2
sum = 0

begin
  new = first + second
  first = second
  second = new

  sum += first if (first % 2).zero?
end while (first < 4_000_000 && second < 4_000_000)

puts sum

File: problem_2/even_fibonacci.js
let initial = 1;
let second = 2;
let result = 0;
let sum;

while (initial < 4000000 && second < 4000000) {
  sum = initial + second;
  initial = second;
  second = sum;

  if (initial % 2 === 0) {
    result += initial;
  }
}

console.log(result);
